client/hooks/useTodos.ts

```diff
@@ -0,0 +1,53 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { fetchTodos, createTodo, updateTodo, deleteTodo } from '../api/todos';
+import type { InsertTodo, UpdateTodo } from '@/shared/todoSchema'; // Importing types
+import { todoInsertSchema, updateTodoSchema } from '@/shared/todoSchema'; // Importing Zod schemas for validation
+
+export const useTodos = () => {
+  const queryClient = useQueryClient();
+
+  // Fetch todos query
+  const todosQuery = useQuery({
+    queryKey: ['todos'],
+    queryFn: fetchTodos,
+  });
+
+  // Mutation for creating a new todo
+  const createTodoMutation = useMutation({
+    mutationFn: async (newTodo: InsertTodo) => {
+      // Validate using Zod schema
+      todoInsertSchema.parse(newTodo);
```

Comment:
we dont need to validate this because we already know its the right type
we only need to validate when recieving untrusted data (e.g. from the network)